### PR TITLE
Fixing keycloak's behavior of treating ConenctionTimeouts and DB exceptions as login error and brute force kickin

### PR DESCRIPTION
**Pull request by pravsjv**

### CODE REVIEW

**Comment by pravsjv** [comment]
Hi Team, Can someone review this.

**Comment by pravsjv** [comment]
Team, can someone have a look and review the changes.

**Comment by pravsjv** [comment]
Adding the JIRA for more description of the issue.
https://issues.redhat.com/browse/KEYCLOAK-18942

**Comment by mvk37** [comment]
Useful bug fixing. If credential couldn't be validated because of connection problem to a backend  brute-force shouldn't count them.

**Review by mvk37 — DISMISSED** [review]

File: federation/ldap/src/main/java/org/keycloak/storage/ldap/idm/store/ldap/LDAPOperationManager.java [hunk 1]
```diff
@@ -536,6 +536,12 @@ public void authenticate(String dn, String password) throws AuthenticationExcept
             }
 
             throw ae;
+        } catch(RuntimeException re){
```
Comment (mposolda) [inline federation/ldap/src/main/java/org/keycloak/storage/ldap/idm/store/ldap/LDAPOperationManager.java:539]: Minor: Is it possible to use spaces in formatting like:
```
} catch (RuntimeException re) {
```
and
```
if (log.isDebugEnabled()) {
```

**Review by mposolda — CHANGES_REQUESTED** [review]
@pravsjv Thanks for the PR! Few points:
- Added one minor inline point for the formatting
- Can you rebase this PR when you address the point? This will allow easier review.
- Can you please add the JIRA number to the commit message? See other messages sent to Keycloak repository for the details.

**Comment by mposolda** [comment]
@pravsjv The PR looks good, but we will need to run the build and I am not able to re-run the build right now. Is it fine for you to rebase this PR on top of latest master? Hopefully that will help to be able to run the PR.

**Comment by mposolda** [comment]
@pravsjv Any update to the previous comment?

**Comment by pravsjv** [comment]
@mposolda Sorry for the delay in responding. Raised the pull request with proper github ID and incorporating the review comments. Here is the pull request for the same - https://github.com/keycloak/keycloak/pull/9561

**Pull request closed without merging**